packages/stock-location-next/src/services/stock-location.ts

```diff
@@ -0,0 +1,194 @@
+  stockLocationAddressService: ModulesSdkTypes.InternalModuleService<any>
+}
+
+const generateMethodForModels = [StockLocationAddress]
+
+/**
+ * Service for managing stock locations.
+ */
+
+export default class StockLocationModuleService<
+  TEntity extends StockLocation = StockLocation,
+  TStockLocationAddress extends StockLocationAddress = StockLocationAddress
+> extends ModulesSdkUtils.abstractModuleServiceFactory<
+  InjectedDependencies,
+  StockLocationTypes.StockLocationDTO,
+  {
+    StockLocation: { dto: StockLocationTypes.StockLocationDTO }
+    StockLocationAddress: { dto: StockLocationTypes.StockLocationAddressDTO }
+  }
+>(StockLocation, generateMethodForModels, entityNameToLinkableKeysMap) {
+  protected readonly eventBusService_: IEventBusService
+  protected baseRepository_: DAL.RepositoryService
+  protected readonly stockLocationService_: ModulesSdkTypes.InternalModuleService<TEntity>
+  protected readonly stockLocationAddressService_: ModulesSdkTypes.InternalModuleService<TStockLocationAddress>
+
+  constructor(
+    {
+      eventBusService,
+      baseRepository,
+      stockLocationService,
+      stockLocationAddressService,
+    }: InjectedDependencies,
+    protected readonly moduleDeclaration?: InternalModuleDeclaration
+  ) {
+    // @ts-ignore
+    super(...arguments)
+
+    this.baseRepository_ = baseRepository
+    this.stockLocationService_ = stockLocationService
+    this.stockLocationAddressService_ = stockLocationAddressService
+    this.eventBusService_ = eventBusService
+  }
+
+  __joinerConfig(): ModuleJoinerConfig {
+    return joinerConfig
+  }
+
+  create(
+    data: CreateStockLocationInput,
+    context: Context
+  ): Promise<StockLocationTypes.StockLocationDTO>
+  create(
+    data: CreateStockLocationInput[],
+    context: Context
+  ): Promise<StockLocationTypes.StockLocationDTO[]>
+
+  /**
+   * Creates a new stock location.
+   * @param data - The input data for creating a Stock Location.
+   * @param context
+   * @returns The created stock location.
+   */
+  @InjectManager("baseRepository_")
+  async create(
+    data: CreateStockLocationInput | CreateStockLocationInput[],
+    @MedusaContext() context: Context = {}
+  ): Promise<
+    StockLocationTypes.StockLocationDTO | StockLocationTypes.StockLocationDTO[]
+  > {
+    const input = Array.isArray(data) ? data : [data]
+
+    const created = await this.create_(input, context)
+
+    const serialized = await this.baseRepository_.serialize<
+      | StockLocationTypes.StockLocationDTO
+      | StockLocationTypes.StockLocationDTO[]
+    >(created, { populate: true })
+
+    return Array.isArray(data) ? serialized : serialized[0]
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async create_(
+    data: CreateStockLocationInput[],
+    @MedusaContext() context: Context = {}
+  ): Promise<TEntity[]> {
+    return await this.stockLocationService_.create(data, context)
+  }
+
+  update(
+    data: UpdateStockLocationInput & { id: string },
```

Comment:
q: the id shouldn't be part of the DTO? also shouldn't we create proper DTO type like for the other modules?